src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-                    continue;
+                    instance_logger.log("There was an error in the upload process and the response is :: " + response,
+                                        "error");
                 }
+                close(sockfd);
             }
+        }
+    }
+    return 0;
+}
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            line = (data);
-
-            if (line.compare(JasmineGraphInstanceProtocol::FILE_RECV_CHK) == 0) {
-                instance_logger.log("Received : " + JasmineGraphInstanceProtocol::FILE_RECV_CHK, "info");
-                write(connFd, JasmineGraphInstanceProtocol::FILE_ACK.c_str(),
-                      JasmineGraphInstanceProtocol::FILE_ACK.size());
-                instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::FILE_ACK, "info");
-            }
-
-            instance_logger.log("File received and saved to " + fullFilePath, "info");
-            loop = true;
+bool JasmineGraphInstanceService::sendFileThroughService(std::string host, int dataPort, std::string fileName,
+                                                         std::string filePath, std::string masterIP) {
+    int sockfd;
+    char data[INSTANCE_DATA_LENGTH + 1];
+    socklen_t len;
+    struct sockaddr_in serv_addr;
+    struct hostent *server;
 
-            Utils::unzipFile(fullFilePath);
-            size_t lastindex = fileName.find_last_of(".");
-            string rawname = fileName.substr(0, lastindex);
-            fullFilePath =
-                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + rawname;
-            std::string aggregatorFilePath =
-                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.aggregatefolder");
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
 
-            DIR *dir = opendir(aggregatorFilePath.c_str());
+    if (sockfd < 0) {
+        std::cerr << "Cannot create socket" << std::endl;
+        return false;
+    }
 
-            if (dir) {
-                closedir(dir);
-            } else {
-                std::string createDirCommand = "mkdir -p " + aggregatorFilePath;
-                FILE *createDirInput = popen(createDirCommand.c_str(), "r");
-                pclose(createDirInput);
-            }
+    server = gethostbyname(host.c_str());
+    if (server == NULL) {
+        std::cerr << "ERROR, no host named " << server << std::endl;
+        return false;
+    }
 
-            std::string copyCommand = "cp " + fullFilePath + " " + aggregatorFilePath;
+    bzero((char *)&serv_addr, sizeof(serv_addr));
+    serv_addr.sin_family = AF_INET;
+    bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
+    serv_addr.sin_port = htons(dataPort);
+    if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
+        std::cerr << "ERROR connecting to port " << dataPort << std::endl;
```

Comment:
Lets use logger here instead of `std::cerr`

Reply:
Removed all occurrences of `std::cerr` in next PR

Reply:
Ok